vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs

```diff
@@ -115,51 +93,130 @@ type private FSharpProjectOptionsReactor (workspace: VisualStudioWorkspaceImpl,
+                    )
+
+                    // while we are not invalidated, we should update the project itself in the cache.
+                    updateProject newProject
+                    return true
+                else
+                    return true
+            else
+                return true
+        }
 
     let rec tryComputeOptionsByFile (document: Document) (ct: CancellationToken) =
         async {
-            let! fileStamp = document.GetTextVersionAsync(ct) |> Async.AwaitTask
-            match singleFileCache.TryGetValue(document.Id) with
+            let isScript = isScriptFile document.FilePath
+            match cache.TryGetValue(document.Project.Id) with
             | false, _ ->
                 let! sourceText = document.GetTextAsync(ct) |> Async.AwaitTask
-                let! scriptProjectOptions, _ = checkerProvider.Checker.GetProjectOptionsFromScript(document.FilePath, sourceText.ToFSharpSourceText())
-                let projectOptions =
-                    if isScriptFile document.FilePath then
-                        scriptProjectOptions
+                let! projectOptions, _ =
+                    if isScript then
+                        checkerProvider.Checker.GetProjectOptionsFromScript(document.FilePath, sourceText.ToFSharpSourceText())
                     else
-                        {
-                            ProjectFileName = document.FilePath
-                            ProjectId = None
-                            SourceFiles = [|document.FilePath|]
-                            OtherOptions = [||]
-                            ReferencedProjects = [||]
-                            IsIncompleteTypeCheckEnvironment = false
-                            UseScriptResolutionRules = SourceFile.MustBeSingleFileProject (Path.GetFileName(document.FilePath))
-                            LoadTime = DateTime.Now
-                            UnresolvedReferences = None
-                            OriginalLoadReferences = []
-                            ExtraProjectInfo= None
-                            Stamp = Some(int64 (fileStamp.GetHashCode()))
+                        async {
+                            return {
+                                ProjectFileName = document.FilePath
+                                ProjectId = None
+                                SourceFiles = [|document.FilePath|]
+                                OtherOptions = [||]
+                                ReferencedProjects = [||]
+                                IsIncompleteTypeCheckEnvironment = false
+                                UseScriptResolutionRules = SourceFile.MustBeSingleFileProject (Path.GetFileName(document.FilePath))
+                                LoadTime = DateTime.Now
+                                UnresolvedReferences = None
+                                OriginalLoadReferences = []
+                                ExtraProjectInfo= None
+                                Stamp = None
```

Comment:
Why is this `None` now?

Reply:
We never used it and it was holding onto an instance that I don't care about.

Reply:
An incremental builder will be rebuilt if a `FSharpProjectOptions`'s `Stamp` is different from the previous, or the previous had no `Stamp`. If neither current or previous have Stamps, it will do a comparison of all the other fields to determine if there is a difference and rebuild an incremental builder.

A `Stamp` change is to indicate a re-evaluation of a project when none of `FSharpProjectOptions`'s other fields have changed. This is not the case with single files or script files, therefore it's fine that we set it to `None`.